benchmarks/src/commonMain/kotlin/BufferOps.kt

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.benchmarks
+
+import kotlinx.benchmark.*
+import kotlinx.io.*
+
+@State(Scope.Benchmark)
+abstract class BufferRWBenchmarkBase {
+    // Buffers are implemented as list of segments, as soon as a segment is empty
+    // it will be unlinked. By reading all previously written data, a segment will be
+    // cleared and recycled, and the next time we will try to write data a new segment
+    // will be requested from the pool. Thus, without having some data in-flight we will
+    // benchmark not only read/write ops performance, but also segments allocation/reclamation.
+    @Param("128")
```

Comment:
Mostly random. I checked other values and there were almost no difference in resulting performance. I left it as a parameter to easily check how it will affect performance in the future.